test_conformance/api/test_api_consistency.cpp

```diff
@@ -473,14 +437,689 @@ int test_consistency_pipes(cl_device_id deviceID, cl_context context,
+
+    const cl_mem_object_type image_types[] = {
+        CL_MEM_OBJECT_IMAGE1D,       CL_MEM_OBJECT_IMAGE1D_BUFFER,
+        CL_MEM_OBJECT_IMAGE2D,       CL_MEM_OBJECT_IMAGE3D,
+        CL_MEM_OBJECT_IMAGE1D_ARRAY, CL_MEM_OBJECT_IMAGE2D_ARRAY,
+    };
+    for (int i = 0; i < ARRAY_SIZE(image_types); i++)
+    {
+        cl_uint numImageFormats = 0;
+        error = clGetSupportedImageFormats(
+            context, CL_MEM_KERNEL_READ_AND_WRITE, image_types[i], 0, NULL,
+            &numImageFormats);
+        test_error(error,
+                   "Unable to query number of CL_MEM_KERNEL_READ_AND_WRITE "
+                   "image formats");
+
+        totalReadWriteImageFormats += numImageFormats;
+    }
+
+    if (maxReadWriteImageArgs == 0)
+    {
+        test_assert_error(
+            totalReadWriteImageFormats == 0,
+            "CL_DEVICE_MAX_READ_WRITE_IMAGE_ARGS returned 0 "
+            "but clGetSupportedImageFormats(CL_MEM_KERNEL_READ_AND_WRITE) "
+            "returned a non-empty set");
+    }
+    else
+    {
+        test_assert_error(
+            totalReadWriteImageFormats != 0,
+            "CL_DEVICE_MAX_READ_WRITE_IMAGE_ARGS is non-zero "
+            "but clGetSupportedImageFormats(CL_MEM_KERNEL_READ_AND_WRITE) "
+            "returned an empty set");
+    }
+
+    return TEST_PASS;
+}
+
+int test_consistency_2d_image_from_buffer(cl_device_id deviceID,
+                                          cl_context context,
+                                          cl_command_queue queue,
+                                          int num_elements)
+{
+    // clGetDeviceInfo, passing CL_DEVICE_IMAGE_PITCH_ALIGNMENT or
+    // CL_DEVICE_IMAGE_BASE_ADDRESS_ALIGNMENT
+    // May return 0, indicating that device does not support Creating a 2D Image
+    // from a Buffer.
+    cl_int error;
+
+    const cl_image_format imageFormat = { CL_RGBA, CL_UNORM_INT8 };
+    const size_t imageDim = 2;
+    const size_t elementSize = 4;
+    const size_t bufferSize = imageDim * imageDim * elementSize;
+
+    clMemWrapper buffer;
+    clMemWrapper image;
+
+    cl_uint imagePitchAlignment = 0;
+    error = clGetDeviceInfo(deviceID, CL_DEVICE_IMAGE_PITCH_ALIGNMENT,
+                            sizeof(imagePitchAlignment), &imagePitchAlignment,
+                            NULL);
+    test_error(error,
+               "Unable to query "
+               "CL_DEVICE_IMAGE_PITCH_ALIGNMENT");
+
+    cl_uint imageBaseAddressAlignment = 0;
+    error = clGetDeviceInfo(deviceID, CL_DEVICE_IMAGE_BASE_ADDRESS_ALIGNMENT,
+                            sizeof(imageBaseAddressAlignment),
+                            &imageBaseAddressAlignment, NULL);
+    test_error(error,
+               "Unable to query "
+               "CL_DEVICE_IMAGE_BASE_ADDRESS_ALIGNMENT");
+
+    bool supports_cl_khr_image2d_from_buffer =
+        is_extension_available(deviceID, "cl_khr_image2d_from_buffer");
+
+    if (imagePitchAlignment == 0 || imageBaseAddressAlignment == 0)
```

Comment:
Probably should test for the extension string here instead.

Reply:
See above.